### DISCUSSION TITLE
Teleport AC's of OTP codes?

### DISCUSSION THREAD

**Discussion by KevtheDev42** [gravitational/teleport#24360] [Q&A] [answered]
Hello everyone!

Iv been met with a peculiar and undesirable situation that I'm trying to improve they only way I'm allowed to. 

I have a generic account I need to share with some people, but I need 2FA turned on using either an Authenticator app or Physical Security Keys, even if we all scan the same QR code or add all of our physical keys I still want the ability to remove some people from the 2FA option. 

I just got teleport up and running on my server just now because I have been meaning to try it out, but I'm wondering if its able to do this? Or any way of tackling this dilemma somehow. 

Thank you! :)

**Comment by webvictim** [comment]
I'm not quite sure I understand the problem statement. Do you mean you want to share one Teleport username/password/OTP between multiple people?

If so, it's possible but I wouldn't recommend it.

If you want to share access to a single login on a server, however, Teleport is designed to do exactly this.

Maybe if you can clarify a little we can give better advice :)

Reading your statement again suggests that maybe you want to share one Teleport username but make the use of 2FA optional for some people on that same account? That isn't possible.

**Comment by KevtheDev42** [comment]
Sorry let me clarify a bit more.

The generic account is completely unrelated to teleport, we received one account from another 3rd party service to be used by several people, the account requires 2FA via either an authenticator app, but we want to implement some access controls on the 2FA option, so that incase a person leaves the project we can revoke their access to the authenticator and change the password at a later time. That being said, I just want to know if Teleport act as a authenticator app so I can administer access to it. 

I know sharing one account between multiple people with the same password isn't ideal but we have no other option, that's why I'm looking into teleport to provide at least some added layer of security. 

Let me know if you have any other questions, thank you again!

**Reply by webvictim** [reply, marked as answer]
Got it, thanks for the clarification. Teleport isn't a broker for second factor authentication as such, but I have a few suggestions.

If there are ways to enforce IP restrictions on this third party app (I'm presuming it's some kind of hosted web service?) then you could potentially set up Teleport application access to proxy access to it and mandate that all
traffic must come from a trusted Teleport IP which you host. Unfortunately it's still going to be fairly unclean from a user perspective as they'd need two sets of credentials - one to log into Teleport, then another to log into the app.

Teleport can broker authentication to apps via JWT if the app supports that, but not many do. Teleport also recently gained the ability to act as a SAML provider and broker authentication that way, although this is an Enterprise-only feature - but many more apps do support SAML authentication.

Outside of this, your best bet is probably to enforce that individual users register their own MFA keys (assuming that the third party site supports them) and then store names against them so you know whose is whose and can revoke the relevant key when the user leaves.

**Reply by KevtheDev42** [reply]
Got it, thank you again!
